### PR TITLE
Adding zip capabilities for downloading an entire work

### PR DESCRIPTION
**Pull request by carolyncole**
Adds rubyzip gem
Add WorkZipService
Modifies download controller to download the work as a zip
Adds view changes to include downloading the work on the work show page

refs #625

### CODE REVIEW

File: app/controllers/downloads_controller.rb [hunk 1]
```diff
@@ -6,20 +6,27 @@ class DownloadsController < ApplicationController
...
       authorize! :read, params[asset_param_key]
     end
+
+    def load_file
+      case asset
+      when GenericWork
+        zip_service = WorkZipService.new(asset, current_ability, work_directory)
+        zip_service.call
+      else
+        super
```
Comment (awead) [inline app/controllers/downloads_controller.rb]: @cam156 what about a guard clause here? Or do you anticipate more cases?

File: app/services/work_zip_service.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+#  This service will only include files the ability can read
+#  The zip file name will be returned from the call method
+#
+# @example
+#   work = GenericWork.find(id) # work with file sets attached
+#
+#   service = WorkZipService.new(work, current_ability)
+#
+#   zip_file_name = service.call
+#
```
Comment (awead) [inline app/services/work_zip_service.rb:14]: yay docs!

File: spec/controllers/downloads_controller_spec.rb [hunk 3]
```diff
@@ -94,15 +94,18 @@
...
+      let(:response) { instance_double ActionDispatch::Response, headers: {} }
 
       before do
         # I must save the work again becuase the factory just sends the representative id to solr
         #  This save sends the id to fedora
         my_work.save
         controller.params[:id] = my_work.id
+        allow(controller).to receive(:response).and_return(response)
       end
       it 'redirects' do
```
Comment (awead) [inline spec/controllers/downloads_controller_spec.rb]: Are you still testing a redirect here, or the zip download?

File: app/services/work_zip_service.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  end
+
+  # create the zip file
+  #
+  def call
+    zipfile_name = File.join(zip_directory, "#{work.title.first.parameterize('_')}.zip")
+
+    Zip::File.open(zipfile_name, Zip::File::CREATE) do |zipfile|
+      work.file_sets.each do |file_set|
+        if file_set.original_file.present? && (ability.can? :read, file_set.id)
```
Comment (awead) [inline app/services/work_zip_service.rb]: You could combine the above two lines into a `select` and remove an indentation level; optional, if you think this is more readable.

**Review by awead — CHANGES_REQUESTED** [review]

**Review by awead — APPROVED** [review]

**Comment by carolyncole** [comment]
@awead this PR is all fixed up.  Coveralls is angry becuase I only added unit tests for the zip service.

**Pull request merged**